x/tunnel/keeper/keeper_packet.go

```diff
@@ -88,6 +65,40 @@ func (k Keeper) ProduceActiveTunnelPackets(ctx sdk.Context) error {
 	return nil
 }
 
+// ProduceActiveTunnelPacket generates a packet and sends it to the destination route for the given tunnel ID.
+// If not enough fund, deactivate the tunnel.
+func (k Keeper) ProduceActiveTunnelPacket(
+	ctx sdk.Context,
+	tunnelID uint64,
+	pricesMap map[string]feedstypes.Price,
+) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic recovered: %v", r)
+			return
+		}
+	}()
```

Comment:
Should we avoid panic in our codebase instead of using try-catch pattern that very anti pattern in Go?